src/backend/gl/src/lib.rs

```diff
@@ -607,7 +610,22 @@ impl adapter::PhysicalDevice<Backend> for PhysicalDevice {
         _usage: image::Usage,
         _view_caps: image::ViewCapabilities,
     ) -> Option<image::FormatProperties> {
-        conv::describe_format(format)?;
+        let conv::FormatDescription {
+            tex_external,
+            tex_internal,
+            data_type,
+            ..
+        } = conv::describe_format(format)?;
+
+        if !self.0.texture_format_filter.check(
+            tex_internal,
+            tex_external,
+            data_type) {
+
+            /* This format is not supported. */
+            return None
```

Comment:
nit: lets not have an early return here, since the rest of the code doesn't have any logic. The if/else should just result in an expression